### PR TITLE
Update Model and UI diagrams in DG

### LINKED ISSUE #250: Model Diagram has outdated model on how meeting is implemented in DG

### PR DESCRIPTION
**Pull request by Pluiexo**
Update with minimal changes, so that the developer guide is still  relevant  with the new model
Closes #250 

Idk how to fix the filtered and all label problem, might want to consider omitting them instead

### CODE REVIEW

File: docs/diagrams/ModelClassDiagram.puml
Comment (whitesnowx) [inline docs/diagrams/ModelClassDiagram.puml:1]: Cant find a fix yet, but can try this, its quite scuffed but maybe improve the diagram


I changed these lines:

`UniquePersonList --> "                       ~* all" Person`

`ModelManager "\n\n\n\n\n\n\n\n\n\n\n\n ~*filtered" <-down-  Person`



File (for copy-paste):
```
@startuml
!include style.puml
skinparam arrowThickness 1.1
skinparam arrowColor MODEL_COLOR
skinparam classBackgroundColor MODEL_COLOR

Package Model as ModelPackage <<Rectangle>>{
Class "<<interface>>\nReadOnlyStaffBook" as ReadOnlyStaffBook
Class "<<interface>>\nReadOnlyMeetingBook" as ReadOnlyMeetingBook
Class "<<interface>>\nReadOnlyUserPrefs" as ReadOnlyUserPrefs
Class "<<interface>>\nModel" as Model
Class StaffBook
Class ModelManager
Class MeetingBook
Class MeetingManager
Class UserPrefs

Class UniqueMeetingList
Class UniquePersonList
Class Person
Class Name
Class Phone
Class Email
Class Module
Class Faculty
Class Venue
Class Tag
Class Availability
Class Meeting

Class I #FFFFFF
}

Class HiddenOutside #FFFFFF
HiddenOutside ..> Model



StaffBook .up.|> ReadOnlyStaffBook
ModelManager .up.|> Model
Model .right.> ReadOnlyUserPrefs
Model .left.> ReadOnlyStaffBook
ModelManager -left-> "1" StaffBook
ModelManager -right-> "1" UserPrefs
UserPrefs .up.|> ReadOnlyUserPrefs

MeetingManager -right-> "1" MeetingBook
MeetingBook .up.|> ReadOnlyMeetingBook
MeetingBook -down->"1" UniqueMeetingList



StaffBook *--> "1" UniquePersonList
UniquePersonList --> "                       ~* all" Person
Person *--> Name
Person *--> Phone
Person *--> Email
Person *--> Module
Person *--> Faculty
Person *--> Venue
Person *--> "*" Tag
Person *--> "*" Availability
Person *--> "1" MeetingManager
UniqueMeetingList *--> "*" Meeting

Person -[hidden]up--> I
UniquePersonList -[hidden]right-> I

Name -[hidden]right-> Phone
Phone -[hidden]right-> Email
Email -[hidden]right-> Module
Module -[hidden]right-> Faculty
Faculty -[hidden]right-> Venue
Venue -[hidden]right-> Tag
Tag -[hidden]right-> Availability

Person -[hidden]right> MeetingManager

ModelManager "\n\n\n\n\n\n\n\n\n\n\n\n ~*filtered" <-down-  Person
@enduml
```

Comment (tsulim) [reply docs/diagrams/ModelClassDiagram.puml:1]: I realize `Favourite` is missing from the model class diagram

Comment (Pluiexo) [reply docs/diagrams/ModelClassDiagram.puml:1]: I can help add `Favourite` as well

**Comment by Pluiexo** [comment]
Will need someone to help me generate the images as my PC is at home

File: docs/diagrams/ModelClassDiagram.puml
Comment (Pluiexo) [reply docs/diagrams/ModelClassDiagram.puml:1]: Model has been fixed

**Review by whitesnowx — APPROVED** [review]
lgtm

**Pull request merged**